source/common/http/http1/codec_impl.cc

```diff
@@ -445,16 +452,21 @@ void ConnectionImpl::completeLastHeader() {
 
   // Check if the number of headers exceeds the limit.
   if (headers_or_trailers.size() > max_headers_count_) {
+    // Crash if not dispatching_, otherwise set error and exit early from the parser.
+    RELEASE_ASSERT(dispatching_, "CodecProtocolError occurred outside of dispatching");
```

Comment:
We must be dispatching during header completion, right? Meaning, I think it's very easy to convince ourselves that this must be the case? If so I'm not sure it's worth this assert here. (vs. encode paths which are a lot more confusing).

Reply:
Hmm yes, `completeLastHeader` has a very tractable call stack, all of which occur in callbacks.